test/command_line_tool_test.cc

```diff
@@ -0,0 +1,231 @@
+#include <cstdio>
+#include <iostream>
+#include <istream>
+#include <regex>
+#include <string>
+#include <sstream>
+
+#include "gtest/gtest.h"
+
+#include "command_line_tool_lib.h"
+
+namespace {
+
+inline uint64_t GetChirpIdFromStr(const std::string &post) {
+  std::smatch m;
+  std::regex_search(post, m, std::regex(" [0-9]*\n"));
+  return stoull(m[0]);
+}
+
+// This series of tests still requires `service_server` and `backend_server` to run simultaneously
+class CommandLineTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    username = "user";
+  }
+
+  std::string username;
+};
+
+// This tests register operation
+TEST_F(CommandLineTest, RegisterTest) {
+  // Set up for redirecting standard output
+  std::stringstream buffer;
+  std::streambuf *old = std::cout.rdbuf(buffer.rdbuf());
+  std::string output;
+
+  // Register one user
+  // ServiceClient::ReturnCodes
+  auto ret = command_tool::Register(username);
+  // This should be successful
+  EXPECT_EQ(ServiceClient::OK, ret);
+  // check whether the output message is correct
+  std::string expected_success_output = std::string("Registered username: ") + username + std::string(": ") + command_tool::service_client.ErrorMsgs[ServiceClient::OK] + std::string("\n");
+  EXPECT_EQ(expected_success_output, buffer.str());
```

Comment:
Just to check the buffer.str() is the string that was just displayed to the console?

Reply:
Yes.